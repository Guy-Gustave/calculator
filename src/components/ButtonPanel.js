import React from 'react';
import Button from './Button';
import '../index.css';

export default function ButtonPanel() {
  const Group1 = ['AC', '+/-', '%', 'รท'];
  const Group2 = ['  7  ', '  8  ', '  9  ', 'x'];
  const Group3 = ['4', '5', '6', '-'];
  const Group4 = ['1', '2', '3', '+'];
  const Group5 = ['0', '.', '='];
  const rightButton = ['รท', 'x', '-', '+', '='];

  const Groups = [Group1, Group2, Group3, Group4, Group5];

  return (
    <div className="calcul">
      {Groups.map((group) => (
        <div className="group" key={group}>
          {group.map((char) => (
            <Button key={char} buttonName={`${char}`}
            color={!rightButton.includes(char)}
            wide={char === '0'}
            clickHandler= {handleClick} />
          ))}
        </div>
      ))}
    </div>
  );
}

ButtonPanel.defaultProps = {
  clickHandler: null,
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};
