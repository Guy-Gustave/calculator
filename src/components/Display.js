import PropTypes from 'prop-types';

const Display = ({ result, next }) => {
  // const { result } = props;
  return (
    <div className="result">
      {next || (result || 0)}
    </div>
  );
};

Display.propTypes = {
  result: PropTypes.string,
  next: PropTypes.string
};

Display.defaultProps = {
  result: '0',
  next: null
};

export default Display;
