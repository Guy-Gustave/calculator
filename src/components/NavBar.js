import React from 'react';
import { Link } from 'react-router-dom';
import '../index.css';
/* eslint-disable */

const NavBar = () => {
  return (
    <div className="navbar">
    <h1>Math Magicians</h1>
    <div className="nav">
      <Link to="/">Home </Link>
      <Link to="/App">Calculator </Link>
      <Link to="/Quote">Quote </Link>
    </div>
  </div>
  
  
  );
};

export default NavBar;
