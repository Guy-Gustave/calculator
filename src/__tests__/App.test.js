import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';
import '../index.css';
import Calculatate from '../logic/calculatate';
import NavBar from '../components/NavBar';


it('renders correctly', () => {
  const tree = renderer
    .create(
      <div className="App">
      <BrowserRouter>
        <NavBar />
        <Display  />
        <ButtonPanel  />
      </BrowserRouter>
    </div>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
