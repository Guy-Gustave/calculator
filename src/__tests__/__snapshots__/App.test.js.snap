// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/App"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>
  <div
    className="result"
  >
    0
  </div>
  <div
    className="calcul"
  >
    <div
      className="group"
    >
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="group"
    >
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="group"
    >
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="group"
    >
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="group"
    >
      <button
        className="gray button text-center text-black button1"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="gray button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange button text-center text-black false"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
